terraform/cloud-platform-components/kiam.tf

```diff
@@ -0,0 +1,190 @@
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P384"
+}
+
+resource "tls_cert_request" "agent" {
+  key_algorithm   = "${tls_private_key.agent.algorithm}"
+  private_key_pem = "${tls_private_key.agent.private_key_pem}"
+
+  subject {
+    common_name = "Kiam Agent"
+  }
+}
+
+resource "tls_locally_signed_cert" "agent" {
+  cert_request_pem   = "${tls_cert_request.agent.cert_request_pem }"
+  ca_key_algorithm   = "${tls_private_key.ca.algorithm}"
+  ca_private_key_pem = "${tls_private_key.ca.private_key_pem}"
+  ca_cert_pem        = "${tls_self_signed_cert.ca.cert_pem}"
+
+  validity_period_hours = 8760 // 1 year
+  early_renewal_hours   = 720  // 1 month
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "client_auth",
+    "server_auth",
+  ]
+}
+
+resource "tls_private_key" "server" {
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P384"
+}
+
+resource "tls_cert_request" "server" {
+  key_algorithm   = "${tls_private_key.server.algorithm}"
+  private_key_pem = "${tls_private_key.server.private_key_pem}"
+
+  subject {
+    common_name = "Kiam Server"
+  }
+
+  dns_names = [
+    "kiam-server",
+  ]
+
+  ip_addresses = [
+    "127.0.0.1",
+  ]
+}
+
+resource "tls_locally_signed_cert" "server" {
+  cert_request_pem   = "${tls_cert_request.server.cert_request_pem }"
+  ca_key_algorithm   = "${tls_private_key.ca.algorithm}"
+  ca_private_key_pem = "${tls_private_key.ca.private_key_pem}"
+  ca_cert_pem        = "${tls_self_signed_cert.ca.cert_pem}"
+
+  validity_period_hours = 8760 // 1 year
+  early_renewal_hours   = 720  // 1 month
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "client_auth",
+    "server_auth",
+  ]
+}
+
+data "template_file" "kiam" {
+  template = "${file("${path.module}/templates/kiam.yaml.tpl")}"
+
+  vars = {
+    kiam_version = "v3.0"
+    ca           = "${base64encode(tls_self_signed_cert.ca.cert_pem)}"
+    agent_cert   = "${base64encode(tls_locally_signed_cert.agent.cert_pem)}"
+    agent_key    = "${base64encode(tls_private_key.agent.private_key_pem)}"
+    server_cert  = "${base64encode(tls_locally_signed_cert.server.cert_pem)}"
+    server_key   = "${base64encode(tls_private_key.server.private_key_pem)}"
+  }
+}
+
+resource "null_resource" "kube_system_kiam_annotation" {
+  provisioner "local-exec" {
+    command = "kubectl annotate --overwrite namespace kube-system 'iam.amazonaws.com/permitted=.*'"
+  }
+}
+
+resource "helm_release" "kiam" {
+  name          = "kiam"
+  chart         = "stable/kiam"
+  namespace     = "kiam"
+  version       = "2.0.0-rc3"
+  recreate_pods = "true"
+
+  values = [
+    "${data.template_file.kiam.rendered}",
+  ]
+
+  depends_on = ["null_resource.deploy"]
+
+  lifecycle {
+    ignore_changes = ["keyring"]
+  }
+}
+
+resource "kubernetes_service" "server-metrics" {
+  metadata {
+    name      = "kiam-server-metrics"
+    namespace = "kiam"
+
+    labels {
+      app       = "kiam"
+      component = "server-metrics"
+    }
+  }
+
```

Comment:
On fresh clusters the `kiam` namespace won't exist and the service creation will fail. Perhaps put a ns creation before.

Reply:
or put a dependency allowing the Helm to run first (and create the ns if it doesn't exist). 

This is the error I'm seeing. Easily remedied by creating the ns manually:
`* kubernetes_service.agent-metrics: namespaces "kiam" not found`

Reply:
@jasonBirchall e9a4931 should fix this but I have not tested it yet

Reply:
This has now been tested.